plugin/cypress/support/commands.ts

```diff
@@ -8,30 +8,27 @@
 // commands please read more here:
 // https://on.cypress.io/custom-commands
 // ***********************************************
-//
-//
-// -- This is a parent command --
-// Cypress.Commands.add('login', (email, password) => { ... })
-//
-//
-// -- This is a child command --
-// Cypress.Commands.add('drag', { prevSubject: 'element'}, (subject, options) => { ... })
-//
-//
-// -- This is a dual command --
-// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
-//
-//
-// -- This will overwrite an existing command --
-// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
-//
-// declare global {
-//   namespace Cypress {
-//     interface Chainable {
-//       login(email: string, password: string): Chainable<void>
-//       drag(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
-//       dismiss(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
-//       visit(originalFn: CommandOriginalFn, url: string, options: Partial<VisitOptions>): Chainable<Element>
-//     }
-//   }
-// }
\ No newline at end of file
+
+declare global {
+    namespace Cypress {
+        interface Chainable {
+            login(OC_CLUSTER_USER: string, OC_CLUSTER_PASS: string, OC_IDP: string): Chainable<void>
+        }
+    }
+}
+
+Cypress.Commands.add('login', (OC_CLUSTER_USER, OC_CLUSTER_PASS, OC_IDP) => {
+    const user = OC_CLUSTER_USER || Cypress.env('OC_CLUSTER_USER')
+    const password = OC_CLUSTER_PASS || Cypress.env('OC_CLUSTER_PASS')
+    const idp = OC_IDP || Cypress.env('OC_IDP')
+
+    cy.visit('/').then(() => {
+        cy.log('OC_IDP: ', typeof (idp), JSON.stringify(idp))
+        if (idp != undefined) {
+            cy.get('.pf-c-button').contains(idp).click()
```

Comment:
I guess that you have used standard Openshift to test PR. In my case I have used local CRC and it works in a different way:
- If you access CRC there is no idp button, direct username/password input types. We should add special idp to avoid line 28 (`undefined` is not possible since there is default value)
- If you deploy local CRC with `yarn start` command there is no login page (like `anonymous` option in Kiali). For this case I would add another special idp value (e.g., `anonymous`) to skip login process.

My 2cents for the method:
```
Cypress.Commands.add('login', (OC_USERNAME, OC_PASSWORD, OC_IDP) => {
  const user = OC_USERNAME || Cypress.env('OC_USERNAME');
  const password = OC_PASSWORD || Cypress.env('OC_PASSWORD');
  const idp = OC_IDP || Cypress.env('OC_IDP');

  if (idp !== 'anonymous') {
    cy.visit('/').then(() => {
      cy.log('OC_IDP: ', typeof idp, JSON.stringify(idp));
      if (idp !== 'crc') {
        cy.get('.pf-c-button').contains(idp).click();
      }
      cy.get('#inputUsername').clear().type(user);
      cy.get('#inputPassword').clear().type(password);
      cy.get('button[type="submit"]').click();
    });
  }
});
```

Probably `crc` is not the best idp. We should investigate how many idp options we could have in Openshift and test them.

Reply:
please see reply in https://github.com/kiali/openshift-servicemesh-plugin/pull/230#discussion_r1445106220, I'm aware for room for improvement.